File: 프로그래머스/3/42884. 단속카메라/단속카메라.java
import java.io.*;
import java.lang.*;
import java.util.*;

class Solution {
    public int solution(int[][] routes) {
        int answer = 0;
        
        Arrays.sort(routes, (a, b) -> {
            if(a[1] == b[1]){
                return Integer.compare(a[0], b[0]);
            }
            return Integer.compare(a[1], b[1]); 
        });
        
        int b = routes[0][1];
        for(int i = 1; i < routes.length; i++){
            int f = routes[i][0];
            
            if(f > b){
                b = routes[i][1];
                answer++;
            }
        }
        
        return answer+1;
    }
}

File: 프로그래머스/3/42884. 단속카메라/단속카메라.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool cmp(const vector<int> &a, const vector<int> &b){
    if(a[1] == b[1])
        return a[0] < b[0];
    return a[1] < b[1];
}

int solution(vector<vector<int>> routes) {
    int answer = 0;
    
    sort(routes.begin(), routes.end(), cmp);
    
    answer = 1;
    int f = routes[0][0];
    int b = routes[0][1];
    for(int i = 1; i < routes.size(); i++){
        if(routes[i][0] > b){
            f = routes[i][0];
            b = routes[i][1];
            answer++;
        }
    }
    return answer;
}